rstdiff/__init__.py

```diff
@@ -944,10 +944,21 @@ def getSectionName(self, node):
             return node["ids"][0]
         return ""  # No idea...
 
+    def getRefinedSectionName(self, node):
+        """Get section name by ignoring user-specified junk words"""
+        node_name = self.getSectionName(node)
+
+        if hasattr(node.document.settings, "ignore_in_section_name"):
+            for word in node.document.settings.ignore_in_section_name:
+                node_name = node_name.replace(word, "")
```

Comment:
Does this handle stuff like the `the` in `Ethereum`, or does that become `Ereum`?

---



Will this do double replacements? Like if `ignore_in_section_name = ["abba", "cge"]` and the input is `"cabbage"`?

---

Should this be case insensitive?

Reply:
I was thinking of replacing entire words without double replacement, so I guess we will need a more sophisticated matching algorithm. I will update using regex. Case insensitive makes more sense.

Reply:
Have updated the algorithm for ignoring words. As per the new algorithm if the the section name is `The Ethereum cabbage c abba ge` :) and the junk words are `["abba", "cge", "the", "ethereum", "c   ge"]` the new section name becomes `cabbage c ge`

This should work well.